docs/DeveloperGuide.md

```diff
@@ -564,17 +563,30 @@ Given below is a series of steps to show how a `search` operation behaves during
 
 3. `Logic.execute()` then calls the `parseCommand`  method of the `GradPadParser` class to parse the string input.
 
-4. `GradPadParser.parseCommand()` identifies the command as a search command and thus creates a `SearchCommand`
-object.
+4. `GradPadParser.parseCommand()` sees that this is an search command, and so uses the `SearchCommandParser`
+class to create a corresponding `SearchCommand`, using the `SearchCommandParser.parse()` method.
 
-5. This command object is then passed back to the `LogicManager` in step 2.
+5. In `SearchCommandParser`, the ModuleCode is first extracted from the string input. 
+
+6. A `SearchCommand` is then created with the ModuleCode, and is passed back to the
+`LogicManager` in step 2. 
```

Comment:
nit: the phrasing for 4 and 6 seems a bit confusing cause both 4 and 6 mention the creation of the `SearchCommand`.

Reply:
changed!